app/code/Magento/MediaContentCatalog/Model/ResourceModel/GetAssetIdsByCategoryStore.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentCatalog\Model\ResourceModel;
+
+use Magento\Catalog\Api\CategoryManagementInterface;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\MediaContentApi\Model\GetAssetIdsByContentFieldInterface;
+use Magento\Store\Api\GroupRepositoryInterface;
+use Magento\Store\Api\StoreRepositoryInterface;
+
+/**
+ * Class responsible to return Asset id by category store
+ */
+class GetAssetIdsByCategoryStore implements GetAssetIdsByContentFieldInterface
+{
+    private const TABLE_CONTENT_ASSET = 'media_content_asset';
+    private const TABLE_CATALOG_CATEGORY = 'catalog_category_entity';
+    private const ENTITY_TYPE = 'catalog_category';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $connection;
+
+    /**
+     * @var StoreRepositoryInterface
+     */
+    private $storeRepository;
+
+    /**
+     * @var GroupRepositoryInterface
+     */
+    private $storeGroupRepository;
+
+    /**
+     * GetAssetIdsByCategoryStore constructor.
+     *
+     * @param ResourceConnection $resource
+     * @param StoreRepositoryInterface $storeRepository
+     * @param GroupRepositoryInterface $storeGroupRepository
+     */
+    public function __construct(
+        ResourceConnection $resource,
+        StoreRepositoryInterface $storeRepository,
+        GroupRepositoryInterface $storeGroupRepository
+    ) {
+        $this->connection = $resource;
+        $this->storeRepository = $storeRepository;
+        $this->storeGroupRepository = $storeGroupRepository;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(string $value): array
+    {
+        $storeView = $this->storeRepository->getById($value);
+        $storeGroup = $this->storeGroupRepository->get($storeView->getStoreGroupId());
+        $categoryIds = $this->getCategoryIdsByRootCategory((int) $storeGroup->getRootCategoryId());
+        $sql = $this->connection->getConnection()->select()->from(
+            ['asset_content_table' => $this->connection->getTableName(self::TABLE_CONTENT_ASSET)],
+            ['asset_id']
+        )->where(
+            'entity_type = ?',
+            self::ENTITY_TYPE
+        )->where(
+            'entity_id IN (?)',
+            $categoryIds
+        );
```

Comment:
Can we simplify the algorithm and abovid additional query to retrieve the category ids?
```suggestion
        )->where(
            'path like ?',
            $storeGroup->getRootCategoryId() . '%'
        );
```

Reply:
Nice catch, I guess it looks way better now, thanks!